### PR TITLE
fix(e2e): properly check errors match

### PR DESCRIPTION
**Pull request by alexfertel**
Follow-up work to #80 

We were previously just checking that an error's signature was contained in the JSON-RPC response, but now we check the revert data fully.

Also polished the API.

### CODE REVIEW

File: lib/e2e/src/assert.rs [hunk 1]
```diff
@@ -1,28 +1,23 @@
-use alloy::{sol_types::SolError, transports::RpcError};
+use alloy::sol_types::SolError;
 
-pub trait Assert<E> {
+pub trait ErrorExt<E> {
     /// Asserts that current error result corresponds to the typed abi-encoded
```
Comment (bidzyyys) [inline lib/e2e/src/assert.rs:4]: No `assert` here; may be misleading.

**Review by bidzyyys — APPROVED** [review]
I like it 🚀 
Left a small comment.

File: lib/e2e/src/assert.rs (hunk 1 above)
Comment (alexfertel) [reply lib/e2e/src/assert.rs:4]: Yeah, I'll rework this a bit, and will probably move this to a separate file.

Comment (bidzyyys) [reply lib/e2e/src/assert.rs:4]: sure, ping me when I should re-review it.

**Review by bidzyyys — APPROVED** [review]
I like the refactor 🚀

**Pull request merged**